gateway/ln-gateway/src/lib.rs

```diff
@@ -1545,6 +1560,236 @@ pub(crate) async fn fetch_lightning_node_info(
     Ok((node_pub_key, alias, network))
 }
 
+impl Gateway {
+    async fn public_key_v2(&self, federation_id: &FederationId) -> Option<PublicKey> {
```

Comment:
I would prefer `gateway_claim_public_key` or something. There's so many public keys, this is going to confuse me